tests/main/snapd-apparmor/task.yaml

```diff
@@ -0,0 +1,73 @@
+summary: Ensure snapd-apparmor works as expected
+
+environment:
+    CONSUMER_SNAP: test-snapd-policy-app-consumer
+
+debug: |
+    journalctl -u snap.apparmor.service
+
+execute: |
+    if ! systemctl is-active snapd.apparmor.service; then
+      echo "Skipping test since snapd.apparmor.service is not active"
+      exit 0
+    fi
+
+    echo "Ensure snapd.apparmor is enabled"
+    systemctl is-enabled snapd.apparmor.service
+
+    # install a test snap which generates a lot of apparmor policies
+    echo "Given a test snap is installed"
+    "$TESTSTOOLS"/snaps-state install-local "$CONSUMER_SNAP"
+    tests.cleanup defer snap remove --purge "$CONSUMER_SNAP"
+
+    # stop snapd so it does not try and load apparmor policies directly
+    echo "And snapd is stopped"
+    systemctl stop snapd.service
+    tests.cleanup defer systemctl restart snapd.service
+
+    # get the current set of profiles but ignore any which may have gotten
+    # created on-the-fly by apparmor for namespaced commands - these
+    # contain / - as these won't get automatically recreated by
+    # snapd.apparmor
+    grep -v / /sys/kernel/security/apparmor/profiles | sort | cut -f1 -d" " > profiles.txt
```

Comment:
nitpick: I cannot think of any case where this could be a problem, since I guess we don't allow tabs (or any other ascii which sorts before `' '`) in profile filenames, but I anyway would find this clear (and robus against future changes) if `sort` was invoked after `cut`.

Reply:
Sure that sounds like a good usability improvement.

Reply:
https://github.com/snapcore/snapd/pull/11129/commits/45a7837f1a46859d66aa5773dbc843644e6f1996